query/src/main/java/org/infinispan/query/impl/massindex/IndexWorker.java

```diff
@@ -75,9 +74,11 @@ private Object extractValue(Object wrappedValue) {
    public Void call() throws Exception {
       preIndex();
       KeyValueFilter filter = getFilter();
-      try (CloseableIterator<CacheEntry<Object, String>> iterator = entryRetriever.retrieveEntries(filter, null, Util.asSet(Flag.CACHE_MODE_LOCAL), null)) {
+      try (Stream<CacheEntry<Object, Object>> stream = cache.getAdvancedCache().withFlags(Flag.CACHE_MODE_LOCAL)
+              .cacheEntrySet().stream()) {
```

Comment:
I wonder how fast or slow is the updateIndex method below?  Adding a parallel stream wouldn't benefit us much if this is not an extremely faster operation.  The other thought is you could refactor this code to use a forEach instead and then a parallel stream could have big benefit, but then again I hope the updateIndex wouldn't block or use IO else this isn't a good fit.